### PR TITLE
docs: add client tools download section

### PR DESCRIPTION
**Pull request by bernardjkim**
Previously removed manual installation methods in https://github.com/gravitational/teleport/pull/29211, but we should still be documenting binary installation methods for downloading client tools.

### CODE REVIEW

File: docs/pages/choose-an-edition/teleport-cloud/downloads.mdx [hunk 1]
```diff
@@ -12,12 +12,63 @@ below.
 Teleport currently maintains DEB and RPM package repositories. Follow these
 instructions to add a package repository and install Teleport.
 
+Binaries include:
+
+Binary Name|Purpose|
+|---|---|
+|`teleport`| The Teleport binary that you can run within your private networks to enable access to resources in your infrastructure|
```
Comment (zmb3) [inline docs/pages/choose-an-edition/teleport-cloud/downloads.mdx:19]: `tsh` and `tctl` are present in the package repos too, right?

**Review by zmb3 — APPROVED** [review]

File: docs/pages/choose-an-edition/teleport-cloud/downloads.mdx [hunk 2]
```diff
@@ -12,12 +12,63 @@ below.
...
+
+The following archives contain the client tool binaries.
+
+Here are the available archives:
+
+Description                        | | | | |
+-----------------------------------|-|-|-|-|
+Linux (Debian/Ubuntu compatible)   | [amd64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-linux-amd64-bin.tar.gz) | [arm64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-linux-arm64-bin.tar.gz) | [armv7](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-linux-arm-bin.tar.gz) | [i386](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-linux-386-bin.tar.gz) |
+Linux (RHEL/CentOS 7.x compatible) | [amd64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-linux-amd64-centos7-bin.tar.gz) |
+macOS                              | [universal amd64/arm64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-darwin-universal-bin.tar.gz) | [amd64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-darwin-amd64-bin.tar.gz) | [arm64](https://cdn.teleport.dev/teleport-v(=cloud.version=)-darwin-arm64-bin.tar.gz) |
```
Comment (webvictim) [inline docs/pages/choose-an-edition/teleport-cloud/downloads.mdx:54]:
```suggestion
macOS                              | [universal amd64/arm64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-darwin-universal-bin.tar.gz) | [amd64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-darwin-amd64-bin.tar.gz) | [arm64](https://cdn.teleport.dev/teleport-ent-v(=cloud.version=)-darwin-arm64-bin.tar.gz) |
```

File: docs/pages/choose-an-edition/teleport-cloud/downloads.mdx (hunk 1 above)
Comment (webvictim) [reply docs/pages/choose-an-edition/teleport-cloud/downloads.mdx:19]: Yes, we should point this out as well so people don't duplicate effort.

File: docs/pages/choose-an-edition/teleport-cloud/downloads.mdx [hunk 3]
```diff
@@ -12,12 +12,63 @@ below.
...
+|`tsh`| A client tool for accessing resources within your cluster|
+
+<Notice title="macOS installations">
+For macOS, `tsh` comes in the main Teleport package and also in a separate tsh-only package.
+The tsh-only package includes a signed variant of `tsh`. Certain functionality, such as Touch ID support,
+only works when `tsh` has been signed. Therefore, if you are installing the main Teleport package on a macOS system,
+we recommend you first install the main Teleport package and then install the tsh-only package over this.
+</Notice>
+
+### Binary installtion
```
Comment (webvictim) [inline docs/pages/choose-an-edition/teleport-cloud/downloads.mdx:44]:
```suggestion
### Binary installation
```

**Review by ptgott — APPROVED** [review]
Minor comments

**Review by webvictim — APPROVED** [review]

**Pull request merged**